### PR TITLE
HasInstance should accept the prototype of external functions.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by LaszloLango** [comment]
Please add "Fixes #1519" to the commit message.

**Comment by zherczeg** [comment]
Done.

File: tests/jerry/function-external.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+try {
+  ({} instanceof assert);
+  assert (false);
+} catch(e) {
+}
```
Comment (LaszloLango) [inline tests/jerry/function-external.js]: I'd add `assert(e instanceof TypeError);` here like in other tests. IMHO the file name should be `regression-test-issue-1519.js`, because this is a bug fix for a GitHub issue.

File: tests/jerry/function-external.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  assert (!({} instanceof assert));
+} catch(e) {
+  assert (false);
+}
+
+try {
+  ({} instanceof Math.sin);
+  assert (false);
+} catch(e) {
+}
```
Comment (LaszloLango) [inline tests/jerry/function-external.js]: `assert(e instanceof TypeError);`

File: tests/jerry/function-external.js (hunk 1 above)
Comment (zherczeg) [inline tests/jerry/function-external.js]: Actually no. The effect will be the same for that test, and that is expected. This patch actually adds a feature.

File: tests/jerry/function-external.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+try {
+  ({} instanceof assert);
+  assert (false);
+} catch(e) {
+}
+
+assert.prototype = {}
+
+try {
+  assert (!({} instanceof assert));
```
Comment (tilmannOSG) [inline tests/jerry/function-external.js]: No space before the opening parenthesis of the function call please since this is JS code.

File: tests/jerry/function-external.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+} catch(e) {
+  assert (false);
```
Comment (tilmannOSG) [inline tests/jerry/function-external.js]: Same here and all the other assert() calls in the test.

File: tests/jerry/regression-test-issue-736.js [hunk 5]
```diff
@@ -31,7 +31,7 @@ try {
 }
 
 try {
-  eval("var x; x instanceof assert;");
+  eval("var x = {}; x instanceof assert;");
```
Comment (tilmannOSG) [inline tests/jerry/regression-test-issue-736.js]: This looks like an unrelated change (bug fix of a test). Shouldn't this be a separate commit? If you discovered the issue after adding the new functionality in this patch then it can be part of this commit but should be mentioned in the commit message.

Comment (zherczeg) [inline tests/jerry/regression-test-issue-736.js]: It is a related test, which fails after this patch since `undefined instanceof assert` does not throw error anymore. This is the correct behaviour. However, the throwing is tested here, so x needs to be an object.

**Comment by zherczeg** [comment]
Thank you for the review, patch updated.

File: tests/jerry/regression-test-issue-736.js (hunk 5 above)
Comment (tilmannOSG) [inline tests/jerry/regression-test-issue-736.js]: Thanks for the clarification!
